polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala

```diff
@@ -787,19 +794,23 @@ class PythonInterpreter private[python] (
 
                       // TODO: can we get better type information from `PyJArray`?
                       // Types that start with "PyJ*" are actually Java values wrapped by jep for use in Python
-                      case other if other.startsWith("PyJ") =>
-                        val jValue = valueAs(classOf[Object])
-                        val typ = runtime.unsafeRun(compiler.reflect(jValue)).symbol.info
-                        (typ, jValue)
+                      case other if other.startsWith("PyJ") => unwrapJavaValue
                       case other =>
-                        val pyObj = valueAs(classOf[PyObject])
-                        if (convert.isDefinedAt((typeStr, pyObj))) {
-                          convert((typeStr, pyObj))
-                        } else {
-                          // should we use the qualified type? It's confusing that both Spark and Pandas have a "DataFrame".
-                          // But in every other case it's just noise.
-                          val typ = appliedType(typeOf[TypedPythonObject[Nothing]].typeConstructor, compiler.global.internal.constantType(Constant(other)))
-                          (typ, new TypedPythonObject[String](valueAs(classOf[PyObject]), runner))
+                        // TODO: is there any way to determine whether it's a wrapped Java value if it's not wrapped in PyJ*?
```

Comment:
@jeremyrsmith FYI, it looks like there was a change in behavior in Jep 4.0.0 - it seems that not all Java objects are wrapped in `PyJ*` wrappers any more. 

[This test](https://github.com/polynote/polynote/blob/master/polynote-kernel/src/test/scala/polynote/kernel/interpreter/python/PythonInterpreterSpec.scala#L448) was failing with an error trying to convert `kernel_stdout` to a `PyObject`. 

It seems the value was left as a `KernelRuntime$Output` - not wrapped in a `PyJObject` - and thus hit the `other =>` case. 

This `try-catch` solution is quite ugly unfortunately I wasn't able to find a better way to handle it.